packages/rules/src/boolean.ts

```diff
@@ -0,0 +1,32 @@
+import { isNullOrUndefined } from '../../shared';
+
+const booleanValidator = (value: unknown) => {
+  if (isNullOrUndefined(value) || !getBoolean(value)) {
+    return false;
+  }
+
+  return !!String(value).trim().length;
+};
+
+function getBoolean(value: unknown) {
+  switch (value) {
+    case true:
+    case 'true':
+    case 'false':
+    case false:
+    case 't':
+    case 'f':
+    case 0:
+    case '0':
+    case 1:
+    case '1':
+    case 'on':
+    case 'off':
+    case 'yes':
+    case 'no':
```

Comment:
That seems an arbitrary list to me, I would assume a `boolean` rule only verifies if the field value is actually `true` or `false`, without casting whatsoever.

For example `on` and `off` while are common values for native HTML checkboxes, they are not "booleans" if that's the case then maybe this rule name should be `checked` instead.